sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

```diff
@@ -39,11 +40,8 @@
      * @throws IllegalArgumentException If {@code key} is {@code null}.
      */
     public Context(Object key, Object value) {
-        if (key == null) {
-            throw new IllegalArgumentException("key cannot be null");
-        }
         this.parent = null;
-        this.key = key;
+        this.key = Objects.requireNonNull(key, "'key' cannot be null");
```

Comment:
nitPick: Like others, can we add fullstop in the end  i.e. "'key' cannot be null."

Reply:
of course :)